router/middleware.go

```diff
@@ -78,3 +87,177 @@ func (h Handlers) CheckLoginMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
 		return next(c)
 	}
 }
+
+func (h Handlers) CheckAdminMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		gob.Register(&User{})
+		sess, err := h.SessionStore.Get(c.Request(), h.SessionName)

```

Comment:
gorilla/sessionsが露出してるの多分おかしくて、`session.Get`を使うべき。Handler構造体にSessionStore入れてるのがそもそも間違ってそう。
```suggestion
		sess, err := session.Get(c, sessionName)

```